mlrun/feature_store/feature_set.py

```diff
@@ -391,6 +403,52 @@ def purge_targets(self, target_names: List[str] = None, silent: bool = False):
             self, mlrun.api.schemas.AuthorizationAction.delete
         )
 
+        purge_targets = self._reload_and_get_status_targets(
+            target_names=target_names, silent=silent
+        )
+
+        if purge_targets:
+            # TODO - benb - run_uuid shouldn't be a part of purge - remove logic after toggle implementation
+            from mlrun.datastore.targets import generate_target_run_uuid
+
+            run_uuid = generate_target_run_uuid()
+            purge_target_names = list(purge_targets.keys())
+            for target_name in purge_target_names:
+                target = purge_targets[target_name]
+                driver = get_target_driver(target_spec=target, resource=self)
+                try:
+                    driver.purge()
+                    driver.run_uuid = run_uuid
+                except FileNotFoundError:
+                    pass
+                del self.status.targets[target_name]
+
+            self.save()
+
+    def update_targets_for_ingest(
+        self,
+        targets: List[DataTargetBase],
+        overwrite: bool = None,
+    ):
+        ingestion_target_names = [t if isinstance(t, str) else t.name for t in targets]
+
+        status_targets = {}
+        if not overwrite:
+            # silent=True always because targets are not guaranteed to be found in status
+            status_targets = (
+                self._reload_and_get_status_targets(
+                    target_names=ingestion_target_names, silent=True
+                )
+                or {}
+            )
+
+        from mlrun.datastore.targets import update_targets_for_ingest
+
+        update_targets_for_ingest(overwrite, targets, status_targets)
```

Comment:
Scanning this PR, this is the only place where the global `update_targets_for_ingest` is called. Other places call the featureset method. Why do we need the global method then? Why not simply put it here (or merge it into this function)?